### PR TITLE
Streamline the memo state

### PR DESCRIPTION
**Pull request by Jim3Things**
- Removed the stale memo state entries from the ExternalRack message
- Removed the redundant rack name from the ExternalRackSummary message (the name is held as the key in the map).  Retained the ExternalRackSummary struct for future expansion.
- Moved the Zone memo fields out of the proto and into the DBInventory struct definition.  This reflects that the memo data is purely internal and in-memory, so there is no need for it to be in the proto.

### CODE REVIEW

File: internal/services/frontend/DBInventory.go [hunk 1]
```diff
@@ -26,6 +26,8 @@ import (
 type DBInventory struct {
 	Mutex sync.Mutex
 	Zone *pb.ExternalZone
+	MaxBladeCount int64
+	MaxCapacity *common.BladeCapacity
```
Comment (MikeCJohnsonUk) [inline internal/services/frontend/DBInventory.go:30]: So for the entire inventory, not just for a zone, assuming their might be more than one zone at some point?

File: internal/services/frontend/DBInventory.go [hunk 2]
```diff
@@ -181,11 +184,10 @@ func (m *DBInventory) GetBlade(rackID string, bladeID int64) (*common.BladeCapac
 // buildSummary constructs the memo-ed summary data for the zone.  This should
 // be called whenever the configured inventory changes.
 func (m *DBInventory) buildSummary() {
-	m.Zone.MaxBladeCount = 0
+	m.MaxBladeCount = 0
```
Comment (MikeCJohnsonUk) [inline internal/services/frontend/DBInventory.go:187]: Strictly speaking, not yet necessary, but what do we envisage for the locking strategy here? call invocation site or target?

File: pkg/protos/inventory/external.proto [hunk 3]
```diff
@@ -38,18 +38,11 @@ message external {
         // specify the blades in the rack.  Each blade is defined by an integer index within that rack, which is used
         // here as the key.
         map<int64, common.blade_capacity> blades = 3;
-
-        // max_blade_capacity is a cached summary of the maximum values for
-        // each of the blade capacity values.  This can be used to size any
-        // collective values, such as having the UI frame the zone visually
```
Comment (MikeCJohnsonUk) [inline pkg/protos/inventory/external.proto:44]: Any chance we could move this comment either to the field defn in dbInventory, or perhaps on the GetMemoDate() func?

Unless it appears somewhere else that I failed to spot/realize.

File: internal/services/frontend/DBInventory.go (hunk 1 above)
Comment (Jim3Things) [reply internal/services/frontend/DBInventory.go:43]: Correct

**Pull request merged**